### PR TITLE
extmod/vfs_fat: add filesize as 4th element of ilistdir tuple

### PR DESCRIPTION
**Pull request by tomlogic**
First pass at a PR for #3405.  Adds a new filesize entry to the `ilistdir()` tuple for platforms using vfs_fat.  Uses the same function to convert `fno.fsize` to an object as the code for stat does.

### CODE REVIEW

File: docs/library/uos.rst [hunk 1]
```diff
@@ -43,18 +43,22 @@ Filesystem access
...
     - *name* is a string (or bytes if *dir* is a bytes object) and is the name of
       the entry;
     - *type* is an integer that specifies the type of the entry, with 0x4000 for
       directories and 0x8000 for regular files;
     - *inode* is an integer corresponding to the inode of the file, and may be 0
       for filesystems that don't have such a notion.
+    - Some platforms may return a 4-tuple that includes the entry's *size*.  For
+      file entries, *size* is an integer representing the size of the file
+      or -1 if unknown.  Its meaning is currently undefined for directory
+      entries, which should return -1.
```
Comment (dpgeorge) [inline docs/library/uos.rst]: It's probably simpler to say that "Its meaning is currently undefined for directory entries.", that's it.  Then a port doesn't need to go to lengths to explicitly put in -1 for directories.  And it leaves it open to add a meaning later on.

Comment (tomlogic) [reply docs/library/uos.rst]: Updated and force pushed.

**Comment by dpgeorge** [comment]
Thanks!

Please note that some vfs_fat tests are failing because the output of ilistdir has changed.  They should be easy to fix.

**Comment by tomlogic** [comment]
I've pushed an update that fixes `listdir()` and the various vfs_fat tests.

**Comment by dpgeorge** [comment]
Thanks for the updates, merged in 4d3a92c67c9b1550eaf07d06ed74de996ee8fa3b

**Pull request closed without merging**